mmv1/third_party/terraform/services/observability/resource_observability_trace_scope_test.go.tmpl

```diff
@@ -0,0 +1,86 @@
+package observability_test
+
+import (
+{{ if ne $.TargetVersionName `ga` -}}
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+        "github.com/hashicorp/terraform-plugin-testing/plancheck"
+{{- end }}
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"testing"
+)
+
+func TestAccObservabilityTraceScope_update(t *testing.T) {
+{{ if ne $.TargetVersionName `ga` -}}
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccObservabilityTraceScope_basic(context),
+			},
+			{
+				ResourceName:            "google_observability_trace_scope.observability_trace_scope",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "trace_scope_id"},
+			},
+			{
+				Config: testAccObservabilityTraceScope_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_observability_trace_scope.observability_trace_scope", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_observability_trace_scope.observability_trace_scope",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "trace_scope_id"},
+			},
+		},
+	})
+{{- end }}
+}
+
+func testAccObservabilityTraceScope_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_observability_trace_scope" "observability_trace_scope" {
+    provider         = google-beta
+    trace_scope_id   = "tf-test-test-scope%{random_suffix}"
+    location         = "global"
+    resource_names   = [
+        "projects/${data.google_project.project.project_id}",
+    ]
+    description      = "A trace scope configured with Terraform"
+}
+
+data "google_project" "project" {
+    provider         = google-beta
+}
+`, context)
+}
+
+func testAccObservabilityTraceScope_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_observability_trace_scope" "observability_trace_scope" {
+    provider         = google-beta
+    trace_scope_id   = "tf-test-test-scope%{random_suffix}"
+    location         = "global"
+    resource_names   = [
+        "projects/${data.google_project.project.project_id}",
```

Comment:
Could you create a new project within the test and also list it here. Basically, this will allow us to test the field update and verify if the API returns the elements in the same order they were sent.